graphql_compiler/schema_transformation/rename_query.py

```diff
@@ -131,18 +169,56 @@ def leave_selection_set(
     def enter_field(
         self, node: FieldNode, key: Any, parent: Any, path: List[Any], ancestors: List[Any]
     ) -> Union[FieldNode, VisitorAction]:
-        """Rename root vertex fields."""
+        """Rename fields."""
         # For a Field to be a root vertex field, it needs to be the first level of
         # selections (fields in more nested selections are normal fields that should not be
         # modified)
         # As FragmentDefinition is not allowed, the parent of the selection must be a query
         # As a query may not start with an inline fragment, all first level selections are
         # fields
         if self.selection_set_level == 1:
-            renamed_node = self._rename_name(node)
-            if renamed_node is node:  # Name unchanged, continue traversal
-                return None
-            else:  # Name changed, return new node, `visit` will make shallow copies along path
-                return renamed_node
+            self.current_type_name.append(node.name.value)
+        else:
+            # Entered a regular field and we want to find its type
+            current_type_name = self.current_type_name[-1]
+            current_type = self.schema.get_type(current_type_name)
+            if not current_type:
+                raise AssertionError(
+                    f"Current type is {current_type_name} which doesn't exist in the schema. This "
+                    f"is a bug."
+                )
+            if not current_type.ast_node:
+                raise AssertionError(
+                    f"Current type {current_type_name} should have non-null field ast_node, which "
+                    f"contains information such as the current type's fields. However, ast_node "
+                    f"was None. This is a bug."
```

Comment:
Here's a practical example of why I suggested the change above. The error message says that `ast_node` was `None` — but we don't actually know that for sure. All we know is that it was falsy, which means it was _probably_ `None`. I've had way too many debugging odysseys over "trivial" differences like this so I'd advise going for the explicit `None` check again:
```suggestion
            if current_type.ast_node is None:
                raise AssertionError(
                    f"Current type {current_type_name} should have non-null field ast_node, which "
                    f"contains information such as the current type's fields. However, ast_node "
                    f"was None. This is a bug."
```